client/src/common/choicescript-compiler.ts

```diff
@@ -0,0 +1,47 @@
+import { URI as Uri } from 'vscode-uri';
+import { IWorkspaceProvider } from './interfaces/vscode-workspace-provider';
+
+export type AllScenesResult = { [key: string]: { labels: { [key: string]: number }, lines: string[] }; }
+export type CompiledChoiceScriptGame = {
+	title: string,
+	scenes: AllScenesResult
+}
+
+export const labelRegex = /^\s*\*label\s+(\w+)/;
+export const titleRegex = /^\s*\*title\s+([\w\s]+)/;
+
+export class ChoiceScriptCompiler {
+
+	constructor(
+		private readonly workspaceProvider: IWorkspaceProvider
+	) {}
+
+	public async compile(scenes: Uri[]): Promise<CompiledChoiceScriptGame>{
+		const allScenes: AllScenesResult = {};
+		let title: string;
+	
+		for (const s of scenes) {
+			const sceneName = s.fsPath.substring(s.fsPath.lastIndexOf('/') + 1, (s.fsPath.length - ".txt".length));
```

Comment:
I figured out: this line is why it's failing on Windows systems. The URIs getting passed on Windows look like:

` c:\\Users\\username\\Documents\\game\\startup`

The code expects them to be Posix-style with forward slashes. We'll need to accommodate both Posix and Windows-style URIs.

Reply:
How sloppy of me!

Reply:
Far too focused on making it work for web and desktop that I wasn't considering different OSes probably.